solr/core/src/java/org/apache/solr/servlet/ResponseUtils.java

```diff
@@ -135,4 +177,10 @@ public static ErrorInfo getTypedErrorInfo(Throwable ex, Logger log) {
     errorInfo.code = code;
     return errorInfo;
   }
+
+  private static boolean hideStackTrace(SolrCore core) {
+    return core != null
+        ? core.getCoreContainer().hideStackTrace()
+        : Boolean.parseBoolean(System.getProperty("solr.hideStackTrace"));
```

Comment:
no need to check the system property value every time you need to log an error. you can move this to the init phase, in the NodeConfig maybe? and pass that value to the log method.

Reply:
@stillalex  : Thanks for your comments.  For some reason, the unit test in this PR is now failing, before I even try to modify anything.  Is anyone aware of changes in the code path of a response that would prevent going though these methods of ResponseUtils?  I'll keep trying, but, no luck after a couple of hours so far.

Reply:
Something similar was done in latest commit.  Thanks.